File: tracsmobile/src/main/java/edu/txstate/mobile/tracs/util/LoginStatus.java
package edu.txstate.mobile.tracs.util;


import java.util.Observable;

/**
 * Really basic helper class to track whether the user is logged in
 * in one spot.
 */
public class LoginStatus extends Observable {
    private static boolean userIsLoggedIn;
    private static LoginStatus loginStatus;

    private LoginStatus() {
        userIsLoggedIn = false; //Just in case Java changes the default
    }

    public static LoginStatus getInstance() {
        if (loginStatus == null) {
            loginStatus = new LoginStatus();
        }
        return loginStatus;
    }

    public boolean isUserLoggedIn() {
        return userIsLoggedIn;
    }


    public void login() {
        if (userIsLoggedIn) {
            return;
        }
        userIsLoggedIn = true;
        updateObservers();
    }

    public void logout() {
        if (!userIsLoggedIn) {
            return;
        }
        userIsLoggedIn = false;
        updateObservers();
    }

    private void updateObservers() {
        loginStatus.setChanged();
        loginStatus.notifyObservers(userIsLoggedIn);
        loginStatus.clearChanged();
    }
}

File: tracsmobile/src/main/java/edu/txstate/mobile/tracs/util/LoginStatus.kt
package edu.txstate.mobile.tracs.util

import java.util.Observable

class LoginStatus : Observable() {
    companion object {
        var userIsLoggedIn: Boolean = false
        @JvmStatic
        val instance: LoginStatus = LoginStatus()
    }


    fun isUserLoggedIn() : Boolean {
        return userIsLoggedIn
    }

    fun login() {
        if (!isUserLoggedIn()) {
            userIsLoggedIn = true
            updateObservers()
        }
    }

    fun logout() {
        if (isUserLoggedIn()) {
            userIsLoggedIn = false
            updateObservers()
        }
    }

    private fun updateObservers() {
        instance.setChanged()
        instance.notifyObservers(isUserLoggedIn())
        instance.clearChanged()
    }
}

